.github/workflows/test.yaml

```diff
@@ -201,22 +201,113 @@ jobs:
         # ${{ github.workspace }}/bin/govendor remove github.com/snapcore/secboot
         # ${{ github.workspace }}/bin/govendor remove +unused
         SKIP_DIRTY_CHECK=1 GO_BUILD_TAGS=nosecboot ./run-checks --unit
+    - name: Archive code coverage data
+      uses: actions/upload-artifact@v2
+      with:
+        name: unit-code-coverage
+        path: .coverage/coverage.out
+    - name: Cache successful run
+      run: |
+        mkdir -p $(dirname "$CACHE_RESULT_STAMP")
+        touch "$CACHE_RESULT_STAMP"
+
+  component-tests:
```

Comment:
about componen test, is it good idea to run in different systems? or just run it in 20.04 is enough? Because if you need better coverage perhaps we could have a spread task to run the component test like we have for unit tests.